pretyping/pretyping.ml

```diff
@@ -1074,8 +1074,11 @@ and pretype_type k0 resolve_tc valcon (env : ExtraEnv.t) evdref lvar = function
 		   evd_comb1 (define_evar_as_sort env.ExtraEnv.env) evdref ev
                | _ -> anomaly (Pp.str "Found a type constraint which is not a type")
            in
-	     { utj_val = v;
-	       utj_type = s }
+           let evkt = e_new_evar env evdref ~src:(loc, knd) ~naming (mkSort s) in
+           let ev,_ = destEvar evkt in
+           evdref := Evd.define ev v !evdref;
+           { utj_val = v;
+	     utj_type = s }
```

Comment:
May I nitpick after the battle here: a comment explaining why we do that would be worthwhile. It is not explained in the commit message, and my natural reaction by reading this piece of code would be to revert it, as it looks semantically like the identity (except for generating a lone dangling evar).